test/JSON/replacerFunction.js

```diff
@@ -0,0 +1,39 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+
+var TEST = function(a, b) {
```

Comment:
curious why not use the unittestframework here?

Reply:
script load stuff requires working path to be the script's folder. These extra 4 lines make my life more easier.

Reply:
RL should take care of that for you

Reply:
On Windows indeed.

Reply:
It doesn't on xplat? I would expect to see a lot more broken.. What changes would we need to make to have a better experience?

UnitTestFramework has a lot of advantages including a uniform way for us to see errors. Generally the only tests that don't use it are ones that need to run at global scope or syntax errors that have different behavior in function scoped evals.

Reply:
>  I would expect to see a lot more broken.. 

No, because `runtests.py` sets the path for javascript file's location.

> What changes would we need to make to have a better experience?

Some cross platform file system / path related improvements to `ch` but I don't see any reason to push this.

Reply:
Fair enough, but new tests should still be using UnitTestFramework unless they can't.

Reply:
Surely there are some tests, they use multiple stuff from UnitTestFramework. I don't see any reason to push it on these tests...

Does it hurt readability ? or performance of test case? or dependability? or reliability? ... Well, this version of these tests help me to try easier and JSON is and probably will be a hot topic for some time.

Reply:
UnitTestFramework has many of the benefits you've listed.

- It's silent in RL, but verbose in console.
- Exceptions explicitly name the type and error message to help regressions where the message changes but the type doesn't
- It helps us group tests together by functionality
- It encourages descriptions of what you're expecting and why it's an interesting test
- Dynamic tests can generate dynamic descriptions
- Asserts are smart about deep comparisons, NaN, etc and provide a centralized location to update with edge cases as the language evolves
- It helps when someone unfamiliar with the test needs to quickly grok what went wrong (e.g., when merging code, someone not on our team / familiar with our product)
- Other JS engines use our unit test framework to run our tests for coverage

E.g.
> assert.areEqual(e3, 3, "First identifier in catch param as pattern is matched and initialized correctly");

Language tests from ES6 onward make extensive use of this framework for the above reasons.

Ultimately it's up to you, but we've had a lot of benefits from using the framework. If it's more than a pushd tests/JSON & ch replacerFunction.js I'd be interested to know.